src/Psy/TabCompletion/Matchers/AbstractMatcher.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Psy\TabCompletion\Matchers;
+
+use Psy\Context;
+use Psy\ContextAware;
+
+/**
+ * Class AbstractMatcher
+ * @package Psy\TabCompletion\Matchers
+ */
+abstract class AbstractMatcher implements ContextAware
+{
+    /** Syntax types */
+    const CONSTANT_SYNTAX = '^[A-Z0-9-_]+$';
+    const VAR_SYNTAX = '^\$[a-zA-Z0-9-_]+$';
```

Comment:
and a constant is not always caps (it is a standard, yes, but still, `define('foo', 'bar'); foo;` **is** valid